ifrs17/Import/Importers.ipynb

```diff
@@ -622,6 +635,75 @@
             "outputs": []
         },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Yield Curves "
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "Import.DefineFormat(ImportFormats.YieldCurve, async (options, dataSet) => {",
+                "\n    Activity.Start();",
+                "\n    var args = GetArgsFromMain(dataSet) with {ImportFormat = ImportFormats.YieldCurve};",
+                "\n    args.ValidateArgsForPeriod();",
+                "\n    if(ApplicationMessage.HasErrors())  return Activity.Finish();",
+                "\n",
+                "\n    var workspace = Workspace.CreateNew();",
+                "\n    workspace.Initialize(x => x.FromSource(DataSource).DisableInitialization<RawVariable>().DisableInitialization<IfrsVariable>());",
+                "\n    var oldYieldCurves = await DataSource.Query<YieldCurve>().ToArrayAsync();",
+                "\n    var hasNameColumn = dataSet.Tables[ImportFormats.YieldCurve].Columns.Where(x => x.ColumnName == nameof(YieldCurve.Name)).Count() > 0;",
+                "\n    var importLog = await Import.FromDataSet(dataSet)",
+                "\n        .WithType<YieldCurve>((dataset, datarow) => {",
+                "\n            var currency = datarow.Field<string>(nameof(YieldCurve.Currency));",
+                "\n            var values = datarow.Table.Columns.Where(c => c.ColumnName.StartsWith(nameof(YieldCurve.Values))).OrderBy(c => c.ColumnName.Length).ThenBy(c => c.ColumnName)",
+                "\n                                .Select(x => datarow.Field<string>(x.ColumnName).CheckStringForExponentialAndConvertToDouble()).ToArray();",
+                "\n            var name = hasNameColumn",
+                "\n                            ? datarow.Field<string>(nameof(YieldCurve.Name))",
+                "\n                            : default(string);",
+                "\n            var item = new YieldCurve {",
+                "\n                Currency = currency,",
+                "\n                Year = args.Year,",
+                "\n                Month = args.Month, ",
+                "\n                Scenario = args.Scenario,",
+                "\n                Values = values,",
+                "\n                Name = name",
+                "\n            };",
+                "\n            return item;",
+                "\n        }).WithTarget(workspace).ExecuteAsync(); ",
+                "\n    ",
+                "\n    if(importLog.Errors.Any()) return Activity.Finish().Merge(importLog); ",
+                "\n    ",
+                "\n    var newYieldCurves = await workspace.Query<YieldCurve>().ToArrayAsync();",
+                "\n    newYieldCurves = newYieldCurves.Except(oldYieldCurves, YieldCurveComparer.Instance()).ToArray();",
```

Comment:
I think what you are trying to do is not that simple, as I think you need to skip things for comparison. I would not have this logic in this PR

Reply:
The logic has been enhanced and a test has been added. 
Further improvements especially about the comparison of the month are expected in the next PR. (@calderinisystemorph)